apps/sms/views/inbox/js/inbox.js

```diff
@@ -715,19 +698,15 @@ var InboxView = {
     var id = record.id;
     var bodyHTML = record.body;
     var thread = Threads.get(id);
-    var draft, draftId;
     var iconLabel = '';
 
     // A new conversation "is" a draft
     var isDraft = thread.isDraft;
 
-    // A an existing conversation "has" a draft
-    // (or it doesn't, depending on the value
-    // returned by thread.hasDrafts)
-    var hasDrafts = thread.hasDrafts;
+    // A an existing conversation has draft.
+    var draft = thread.getDraft();
 
-    if (hasDrafts) {
-      draft = Drafts.byThreadId(thread.id).latest;
+    if (!isDraft && draft) {
       timestamp = Math.max(draft.timestamp, timestamp);
       // If the draft is newer than the message, update
       // the body with the draft content's first string.
```

Comment:
note: this is what we want to change in https://bugzilla.mozilla.org/show_bug.cgi?id=958105